index.js

```diff
@@ -87,7 +88,7 @@ export function trough() {
      *
      * @template MainOutputValue, RestOutputValue
      * @param {Error?} error
-     * @param {[MainOutputValue] | [MainOutputValue, ...RestOutputValue[]]} [output]
+     * @param {[MainOutputValue?, ...RestOutputValue[]]} [output]
```

Comment:
rest types of 0-n in length.
From usage elsewhere it appears `MainOutputValue` is optional?